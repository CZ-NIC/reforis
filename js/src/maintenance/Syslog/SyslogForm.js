/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";
import { CheckBox, ForisURLs } from "foris";

SyslogForm.defaultProps = {
    formErrors: {},
};

SyslogForm.propTypes = {
    formErrors: PropTypes.shape({
        noDiskMounted: PropTypes.string,
    }),
    formData: PropTypes.shape({
        persistent_logs: PropTypes.bool,
    }),
    setFormValue: PropTypes.func,
};

export default function SyslogForm({ formData, setFormValue, formErrors }) {
    return (
        <>
            <h2>{_(`System logs retention`)}</h2>
            <p
                dangerouslySetInnerHTML={{
                    __html: _(
                        `During device reboot, system logs are lost by default. This option allows to save system logs using Storage plugin. Logs are saved into /srv/log/messages. To enable this option, you have to configure a storage in the <a href="${ForisURLs.storage}" title="Go to Storage plugin">Storage plugin</a> first.`
                    ),
                }}
            />
            <CheckBox
                label={_("Enable system logs retention")}
                disabled={!!formErrors.noDiskMounted}
                checked={formData.persistent_logs}
                onChange={setFormValue((value) => ({
                    persistent_logs: { $set: value },
                }))}
            />
        </>
    );
}
