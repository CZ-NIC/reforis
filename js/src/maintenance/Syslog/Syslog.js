/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { ForisForm } from "foris";
import { API_MODULE_URLs } from "../../common/API";
import SyslogForm from "./SyslogForm";

export default function Syslog() {
    return (
        <ForisForm
            forisConfig={{
                endpoint: API_MODULE_URLs.storage,
            }}
            prepData={prepData}
            prepDataToSubmit={prepDataToSubmit}
            validator={validator}
        >
            <SyslogForm />
        </ForisForm>
    );
}

function prepData(formData) {
    return formData;
}

function prepDataToSubmit(formData) {
    delete formData.disk_mounted;
    delete formData.formating;
    delete formData.old_device;
    delete formData.old_uuid;
    delete formData.state;
    delete formData.uuid;

    return formData;
}

function validator(formData) {
    const errors = {};
    if (formData.disk_mounted !== true) {
        errors.noDiskMounted = _("A drive has to be configured first.");
    }
    return errors.noDiskMounted ? errors : undefined;
}
