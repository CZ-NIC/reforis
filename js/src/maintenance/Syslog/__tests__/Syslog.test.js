/*
 * Copyright (C) 2021 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import {
    fireEvent,
    getByText,
    render,
    wait,
} from "foris/testUtils/customTestRender";
import Syslog from "../Syslog";
import syslogFixture from "./__fixtures__/syslog";
import mockAxios from "jest-mock-axios";

describe("Syslog", () => {
    let syslogContainer;

    beforeEach(async () => {
        const { container } = render(<Syslog />);
        mockAxios.mockResponse({ data: syslogFixture });
        await wait(() => getByText(container, "System logs retention"));
        syslogContainer = container;
    });

    it("Snapshot form enabled", async () => {
        expect(syslogContainer).toMatchSnapshot();
    });

    it("Snapshot form disabled", async () => {
        const { container } = render(<Syslog />);
        mockAxios.mockResponse({ data: { disk_mounted: false } });
        await wait(() => getByText(container, "System logs retention"));
        syslogContainer = container;
        expect(syslogContainer).toMatchSnapshot();
    });

    it("Success post request", () => {
        fireEvent.click(getByText(syslogContainer, "Save"));
        expect(mockAxios.post).toHaveBeenCalledWith(
            "/reforis/storage/api/settings",
            { persistent_logs: true },
            expect.anything()
        );
    });
});
