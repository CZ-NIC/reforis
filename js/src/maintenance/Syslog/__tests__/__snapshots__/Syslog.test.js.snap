// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Syslog Snapshot form disabled 1`] = `
<div>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        System logs retention
      </h2>
      <p>
        During device reboot, system logs are lost by default. This option allows to save system logs using Storage plugin. Logs are saved into /srv/log/messages. To enable this option, you have to configure a storage in the 
        <a
          href="/reforis/storage"
          title="Go to Storage plugin"
        >
          Storage plugin
        </a>
         first.
      </p>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-checkbox "
        >
          <input
            class="custom-control-input"
            disabled=""
            id="1"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="1"
          >
            Enable system logs retention
          </label>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
          disabled=""
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Syslog Snapshot form enabled 1`] = `
<div>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        System logs retention
      </h2>
      <p>
        During device reboot, system logs are lost by default. This option allows to save system logs using Storage plugin. Logs are saved into /srv/log/messages. To enable this option, you have to configure a storage in the 
        <a
          href="/reforis/storage"
          title="Go to Storage plugin"
        >
          Storage plugin
        </a>
         first.
      </p>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-checkbox "
        >
          <input
            checked=""
            class="custom-control-input"
            id="1"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="1"
          >
            Enable system logs retention
          </label>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
